lib/fizzy/execute.cpp

```diff
@@ -840,7 +840,7 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, const Value* args,
                     throw std::bad_alloc();
                 memory->resize(new_pages * PageSize);
             }
-            catch (std::bad_alloc const&)
+            catch (...)
```

Comment:
Clang-tidy complains it can't be `noexcept` without this. I guess it assumes that `resize` call above can throw any exception.

Reply:
Isn't it that we are not catching `std::length_error` of https://en.cppreference.com/w/cpp/string/basic_string/resize ?

Reply:
Looks like catching `bad_alloc` and `length_error` is enough to silence the warning.

Logically `std::bad_array_new_length` (which would mean trying to allocate more than max size_t memory) can't happen both because one element of string is 1 byte and number of elements is given as size_t, but also because `std::length_error` is supposed to be thrown if new length is more than `string::max_size()`.

Reply:
It think it's correct only to return `-1` because `memory.grow` can't trap according to the spec.

Reply:
Btw, what does `resize` do if the old and new size equal? Does it have an early exit?

Reply:
Asking because querying the current size can be done via `memory.grow 0` in wasm, so that may be a "common" pattern.

Reply:
It would reallocate only if new size is greater than `capacity()`, so yeah, should return quickly.

Reply:
Fixed in #737.